### PR TITLE
Topic/iterators

### PR DESCRIPTION
**Pull request by godunko**

### CODE REVIEW

File: source/text/vss-strings-cursors-iterators.ads [hunk 1]
```diff
@@ -45,6 +45,10 @@ package VSS.Strings.Cursors.Iterators is
      abstract limited new VSS.Strings.Cursors.Abstract_Segment_Cursor
        with private;
 
+   function Forward
+     (Self : in out Abstract_Segment_Iterator) return Boolean is abstract;
+   --  Move cursor to the next line
```
Comment (reznikmm) [inline source/text/vss-strings-cursors-iterators.ads:50]: s/next line/next segment/ ?

**Review by reznikmm — DISMISSED** [review]
Approved with a few comments

File: source/text/vss-unicode.ads [hunk 2]
```diff
@@ -38,15 +38,21 @@ package VSS.Unicode is
      new Interfaces.Unsigned_32 range 16#00_0000# .. 16#10_FFFF#;
 
    type UTF8_Code_Unit is mod 2 ** 8;
-   type UTF8_Code_Unit_Count is new Natural;
+   type UTF8_Code_Unit_Offset is new Interfaces.Integer_32;
```
[the remark this answers is not in the document]
Comment (godunko) [reply source/text/vss-unicode.ads:41]: Integer_32 is already defined type, I don't need to remember range and check for real size ;)

File: source/text/implementation/vss-implementation-string_handlers.adb [hunk 3]
```diff
@@ -27,6 +27,29 @@ package body VSS.Implementation.String_Handlers is
 
    use type VSS.Unicode.Code_Point;
 
+   ------------
```
[the remark this answers is not in the document]
Comment (godunko) [reply source/text/implementation/vss-implementation-string_handlers.adb:30]: Why not?

File: source/text/vss-strings-cursors-iterators.ads (hunk 1 above)
Comment (godunko) [reply source/text/vss-strings-cursors-iterators.ads]: Fixed

**Review by reznikmm — APPROVED** [review]

**Pull request merged**